iroh-p2p/benches/lru_cache.rs

```diff
@@ -0,0 +1,316 @@
+//! Test the LRU cache implementation.
+//!
+//! These are a few simple tests of the operations we do on empty and full caches.  Mostly
+//! how populated the cache is doesn't seem to affect things much.
+//!
+//! # Running the benchmarks
+//!
+//! Install `cargo-criterion`:
+//!
+//! ```shell
+//! cargo install cargo-criterion
+//! ```
+//!
+//! Run the benchmarks:
+//!
+//! ```shell
+//! cargo criterion -p iroh-p2p
+//! ```
```

Comment:
ran these locally, works! If we add more benchmarks that use this pattern we can define an `xtask` to run them. I'd skip it for this one, as we're mainly hanging on to this bench for future use

Reply:
I am confused, you can just run the benchmarks with `cargo bench` if the config is setup correctly: https://github.com/bheisler/criterion.rs#quickstart

Reply:
true, i forgot that worked by the time i was finished: https://github.com/n0-computer/iroh/pull/483